src/main/java/com/terraforged/mod/server/command/TerraCommand.java

```diff
@@ -309,10 +335,37 @@ private static CommandSyntaxException createException(String type, String messag
 
     private static ITextComponent createTeleportMessage(BlockPos pos) {
         return TextComponentUtils.wrapInSquareBrackets(new TranslationTextComponent(
-                "chat.coordinates", pos.getX(), "~", pos.getZ()
+                "chat.coordinates", pos.getX(), pos.getY(), pos.getZ()
         )).applyTextStyle((style) -> style.setColor(TextFormatting.GREEN)
                 .setClickEvent(new ClickEvent(ClickEvent.Action.SUGGEST_COMMAND, "/tp @s " + pos.getX() + " " + pos.getY() + " " + pos.getZ()))
                 .setHoverEvent(new HoverEvent(HoverEvent.Action.SHOW_TEXT, new TranslationTextComponent("chat.coordinates.tooltip")))
         );
     }
+
+    private static ITextComponent createIdentifierMessage(int identifier) {
+        return new StringTextComponent("") // Gotta make sure parent style is default
+                   .appendSibling(TextComponentUtils.wrapInSquareBrackets(new StringTextComponent(
+                        String.format("%d", identifier)
+                   )).applyTextStyle(TextFormatting.GOLD));
+    }
+
+    private static ITextComponent createTerrainMessage(Terrain terrain) {
+        return new StringTextComponent("") // Gotta make sure parent style is default
+                   .appendSibling(new StringTextComponent(terrain.getName()).applyTextStyle(TextFormatting.DARK_GREEN));
+    }
+
+    private static ITextComponent createBiomeDisplayMessage(Biome biome) {
+        return new StringTextComponent("") // Gotta make sure parent style is default
+                   .appendSibling(new StringTextComponent(biome.getDisplayName().getFormattedText()).applyTextStyle(TextFormatting.DARK_GREEN));
```

Comment:
Can you not format the TextComponent returned by `getDisplayName()` rather than rendering the string and wrapping it up again?